Bump Geant4 to v11.0.3

ihrivnac opened the pull request:
The patch03 includes fixes for the problem with hyper-nuclei decays  reported in JIRA  O2-3223.

ihrivnac commented:
@TimoWilken ,  the build/alidist/O2Suite/o2/macOS  is waiting for 2 days. Is it normal ? Thank you.

ktf commented:
There is an issue with:

```
CMake Error at /sw/BUILD/0a7f36f1bbef40e43dec066f6c6058c1951ed77b/GEANT4/Externals/G4ABLA-3.1/src/G4ABLA-stamp/download-G4ABLA.cmake:170 (message):
  Each download failed!

    error: downloading 'https://cern.ch/geant4-data/datasets/G4ABLA.3.1.tar.gz' failed
          status_code: 35
          status_string: "SSL connect error"
          log:
          --- LOG BEGIN ---
            Trying 2001:1458:201:70::100:26:443...

  Connected to cern.ch (2001:1458:201:70::100:26) port 443 (#0)
```

is this real?

ihrivnac commented:
> There is an issue with:
> 
> ```
> CMake Error at /sw/BUILD/0a7f36f1bbef40e43dec066f6c6058c1951ed77b/GEANT4/Externals/G4ABLA-3.1/src/G4ABLA-stamp/download-G4ABLA.cmake:170 (message):
>   Each download failed!
> 
>     error: downloading 'https://cern.ch/geant4-data/datasets/G4ABLA.3.1.tar.gz' failed
>           status_code: 35
>           status_string: "SSL connect error"
>           log:
>           --- LOG BEGIN ---
>             Trying 2001:1458:201:70::100:26:443...
> 
>   Connected to cern.ch (2001:1458:201:70::100:26) port 443 (#0)
> ```
> 
> is this real?

This data set did not change since the last G4 version, so it seems to be a temporary network problem; I downloaded the tar file for the URL without any problem.

ktf commented:
Not sure what is going on with one of the mac builders. I took for good the ARM build and I merged this.
